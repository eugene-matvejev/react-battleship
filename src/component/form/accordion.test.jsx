import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Accordion from './accordion';

configure({ adapter: new Adapter() });

describe('<Accordion/>', () => {
    const props = {
    };

    describe('render', () => {
        it('with default/required props', () => {
            const c = shallow(<Accordion {...props} />);

            expect(c).toMatchSnapshot();
        });

        describe('with optional props', () => {
            [
                ['className', '{{className}}'],
                ['title', '{{title}}'],
                ['data-cy', '{{data-cy}}'], /** should be passed 'as is' */
                ['isCollapsed', true],
                ['isCollapsed', false],
            ].forEach(([prop, v]) => {
                it(`[::${prop}] as "${v}"`, () => {
                    const c = shallow(<Accordion {...props} {...{ [prop]: v }} />);

                    expect(c).toMatchSnapshot();
                });
            });
        });
    });
});
