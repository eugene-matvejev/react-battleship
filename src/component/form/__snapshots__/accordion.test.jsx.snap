// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Accordion/> render with default/required props 1`] = `
<section
  className="accordion "
/>
`;

exports[`<Accordion/> render with optional props [::className] as "{{className}}" 1`] = `
<section
  className="accordion {{className}}"
/>
`;

exports[`<Accordion/> render with optional props [::data-cy] as "{{data-cy}}" 1`] = `
<section
  className="accordion "
/>
`;

exports[`<Accordion/> render with optional props [::isCollapsed] as "false" 1`] = `
<section
  className="accordion "
/>
`;

exports[`<Accordion/> render with optional props [::isCollapsed] as "true" 1`] = `
<section
  className="accordion--collapsed "
/>
`;

exports[`<Accordion/> render with optional props [::title] as "{{title}}" 1`] = `
<section
  className="accordion "
>
  <h2
    className="accordion_title"
  >
    {{title}}
  </h2>
</section>
`;
